ladybug_geometry/geometry3d/face.py

```diff
@@ -1151,6 +1151,94 @@ def sub_faces_by_ratio_sub_rectangle(self, ratio, sub_rect_height, sill_height,
+                the rectangle to start the sub-rectangles. Note that, if the
+                ratio is too large for the height, the ratio will take precedence
+                and the sub-rectangle height will be smaller than this value.
+            rect_height: A number for the target height of the output rectangles. Note that, if the ratio is too large for the height,
+                the ratio will take precedence and the sub-rectangle height will
+                be larger than this value.
+            changed_var: string indicating the variable being changed 
+
+        Returns:
+            A list of Face3D objects for sub faces.
+        """
+        sub_face_height_final = rect_height if sub_face_height > rect_height else sub_face_height
+        sill_height_final = 0 if sill_height < 0.01 * rect_height else rect_height-0.1 if sill_height > rect_height-0.1 else sill_height
+
+        if sub_face_height_final + sill_height_final > rect_height:
+            if changed_var == "sill_height_value":
+                sub_face_height_final = rect_height - sill_height_final
+            else:
+                sill_height_final = rect_height - sub_face_height_final
+
+        max_width_break_up = wall_length/2
+        num_divisions = round(wall_length/dist_breakup) if wall_length > dist_breakup/2 else 1
+        if sub_face_width < max_width_break_up and num_divisions * sub_face_width <= wall_length and num_divisions != 1:
+            div_dist = wall_length/2 if num_divisions == 1 else dist_breakup
+            if num_divisions * sub_face_width + (num_divisions-1) * (dist_breakup - sub_face_width) > wall_length:
+                num_divisions = math.floor(wall_length/dist_breakup)
+
+            is_odd = lambda num: num % 2
+            start_pt_x = (num_divisions / 2) * div_dist if is_odd(num_divisions) == 0 else (math.floor(num_divisions / 2) + 0.5) * div_dist
+
+            btm_div_pts = [[start_pt_x, 0, sill_height_final]]
+            remainder = wall_length - (div_dist * num_divisions)
+            total_dist = remainder/2
+
+            while total_dist < wall_length:
+                total_dist += div_dist
+                next_pt = (wall_length / 2) - total_dist
+                btm_div_pts.append([next_pt, 0, sill_height_final])
+
+            sub_face_lines_start = [[btm_div_pts[i], btm_div_pts[i+1]] for i in range(num_divisions)]
+
+            line_cent_pt = [[line[1][0]+((line[0][0]-line[1][0])/2), 0, line[0][2]]
+                            for line in sub_face_lines_start]
+            dist_cent_line = div_dist if num_divisions != 1 else wall_length
+
+            sub_face_line_scale = sub_face_width / div_dist
+
+            sub_face_lines_start = [[[x[0][0] - ((x[0][0]-y[0])*(1-sub_face_line_scale)), 0, x[0][2]],[x[1][0] + ((y[0]-x[1][0])*(1-sub_face_line_scale)), 0, x[0][2]]]
+                                    for x, y in zip(sub_face_lines_start, line_cent_pt)]
```

Comment:
I would recommend using other ladybug_geometry objects here like `LineSegment3Ds`, which you can then extrude into `Face3D`s that you can return:
https://github.com/ladybug-tools/ladybug-geometry/blob/master/ladybug_geometry/geometry3d/line.py#L9